pants.ini

```diff
@@ -414,11 +414,19 @@ config_path: src/docs/docsite.json
 toolchain_variant: gnu
 
 [native-build-step.cpp-compile-settings]
+# TensorFlow 1.13 on python 3.7 specifically uses a newer C++ ABI than any other TensorFlow release,
+# so in this repo's CI we override this option to avoid using the previous ABI for our python 3.7
+# testing by setting PANTS_NATIVE_BUILD_STEP_CPP_COMPILE_SETTINGS_COMPILER_OPTION_SETS="[]"
```

Comment:
Rather than setting this in `.travis.yml`, would it be possible to temporarily modify `os.env` in the `test_zero_out_op.py` file to set this env var? I'd love where possible to keep this issue as locally scoped as possible.

For example, use `with environment_as( PANTS_NATIVE_BUILD_STEP_CPP_COMPILE_SETTINGS_COMPILER_OPTION_SETS=[]):` from `contextutil.py`.

Reply:
In fact, could all 3 of these entries in `pants.ini` be moved into `test_zero_out_op.py` by chance with `PANTS_` env vars? I wouldn't think when looking at `test_zero_out_op.py` to ever look at `pants.ini` for values that were impacting it, which is surprising.

Reply:
This test is currently only run as part of the testprojects integration, not inside any other integration test, so I'm pretty sure editing the environment inside the python file like you describe won't affect the pants options used unless I'm missing something, as the pants options would already be set long before executing the test.

If we were to make it into its own integration test (not a terrible idea), we could make sure to change the default option sets to `[]` in the config we pass to the pants run. However, that would mean the testproject still fails on python 3.7, so we would have to add it to some blacklist. Having a comment with a TODO in the python 3.7 shards (a comment which I would add to this PR once #7261 is merged) seems like a more focused change than adding this target to a blacklist. Let me know what you think.